server/resources/migrations/68_attrs_etype_label_schema.up.sql

```diff
@@ -0,0 +1,46 @@
+ALTER TABLE attrs ADD COLUMN etype text;
+ALTER TABLE attrs ADD COLUMN label text;
+ALTER TABLE attrs ADD COLUMN reverse_etype text;
+ALTER TABLE attrs ADD COLUMN reverse_label text;
+
+ALTER TABLE attrs
+ADD CONSTRAINT attrs_etype_label_unique UNIQUE (app_id, etype, label);
+
+ALTER TABLE attrs
+ADD CONSTRAINT attrs_reverse_etype_label_unique UNIQUE (app_id, reverse_etype, reverse_label);
+
+CREATE OR REPLACE FUNCTION check_attrs_unique_names ()
```

Comment:
Should this part go in part 2? I think you'll hit errors with the null values.

Reply:
No, nulls don’t count in unique

Reply:
Oh right, null won't equal null here.